modules/apps/27-interchain-accounts/controller/keeper/account.go

```diff
@@ -21,13 +21,17 @@ func (k Keeper) RegisterInterchainAccount(ctx sdk.Context, connectionID, owner s
 		return err
 	}
 
-	if k.portKeeper.IsBound(ctx, portID) {
-		return sdkerrors.Wrap(icatypes.ErrPortAlreadyBound, portID)
+	// if there is an active channel for this portID / connectionID return an error
+	activeChannelID, found := k.GetOpenActiveChannel(ctx, connectionID, portID)
+	if found {
+		return sdkerrors.Wrapf(icatypes.ErrActiveChannelAlreadySet, "existing active channel %s for portID %s on connection %s for owner %s", activeChannelID, portID, connectionID, owner)
 	}
 
-	cap := k.BindPort(ctx, portID)
-	if err := k.ClaimCapability(ctx, cap, host.PortPath(portID)); err != nil {
-		return sdkerrors.Wrap(err, "unable to bind to newly generated portID")
+	if !k.IsBound(ctx, portID) {
```

Comment:
I switched this to use our custom keeper fn `IsBound` which also ensures the capability has been claimed for this portID. Otherwise, we would panic below (the tests picked this up).

Reply:
This can still panic. The purpose of using the `portKeeper.IsBound` was that ports claimed outside of ics27 (by another IBC module) would cause a panic here since you are only checking the portID's claimed by the controller keeper and not the portID's binded to via 05-port. 

The code needs to be more nuanced:
```go
switch:
case portkeeper.IsBound() && !k.IsBound():
    return err (controller keeper doesn't own capability)
    
case !portKeeper.IsBound():
    k.BindPort
    k.ClaimCapability
    
}
```
Or something like that

Reply:
On second thought, It might be better to panic on the case that the controller doesn't own the capability since this means another module on the chain claimed a capability in the namespace of ics27. No real difference since a panic just results in a failed transaction anyways. A panic might just be more likely to surface the problem (we should likely investigate closer if that error ever did occur)

Reply:
we should still use port keeper `IsBound()` though

Reply:
So, leave it as is then?

Reply:
No, the current code unintentionally has ok behaviour. If the port is bounded by a different moudle, it will panic on `BindPort()`, but as far as I can tell this is unintentional. I'd prefer to restructure the code such that we have clearly defined expected paths such that if anything changes in the future, we aren't resting the correctness of our code on assumptions (what if BindPort returns an error instead of panic'ing)

I would probably use my switch above, but panic instead of returning an error

Reply:
Alright, gotcha. Thanks for making it clear :+1: